geomstats/geometry/invariant_metric.py

```diff
@@ -387,10 +391,12 @@ def log(self, point, base_point=None):
             Tangent vector at the base point equal to the Riemannian logarithm
             of point at the base point.
         """
+        identity = gs.to_ndarray(self.group.identity, to_ndim=2)
         if base_point is None:
-            base_point = self.group.identity
+            base_point = identity
         base_point = self.group.regularize(base_point)
-        if gs.allclose(base_point, self.group.identity):
+
+        if gs.allclose(base_point, identity):
```

Comment:
Did you check that this works when there are n > 1 base points ? I think I ran into errors with pytorch or tf in those cases

Reply:
Not yet. `invariant_metric` is actually not refactorized in this PR: I only did SO and SE to keep the PR reasonable, just fixing what was breaking in invariant_metric for it.

The next PR will focus on invariant_metric.